lib/ajv.d.ts

```diff
@@ -1,13 +1,9 @@
-declare var ajv: {
-  (options?: ajv.Options): ajv.Ajv;
-  new (options?: ajv.Options): ajv.Ajv;
-  ValidationError: ValidationError;
-  MissingRefError: MissingRefError;
-  $dataMetaSchema: object;
-}
+declare function ajv(options?: ajv.Options): ajv.Ajv;
```

Comment:
I think that the usage as constructor may have been lost? Also the issue is to export only ErrorParameters interfaces to use them as type guards - why all interfaces should be exported?

Reply:
> I think that the usage as constructor may have been lost?

The `declare function ajv(options?: ajv.Options): ajv.Ajv;` and `export = ajv` in the file ensure that the constructor type isn't lost.

> Also the issue is to export only ErrorParameters interfaces to use them as type guards - why all interfaces should be exported?

Unfortunately, the only way I found to combine the `declare function ajv(...)` type with the `namespace ajv` type was to export all the types inside. Otherwise the types themselves aren't exported as well. 

I tried various ways of fixing the issue, but the only way that worked and didn't break anything was following the typings guide https://github.com/typings/typings/blob/master/docs/examples.md#function--utility here, which says to declare `function` and `namespace` types (where all the types you need from the namespace are exported).